### PR TITLE
fix: improve alchemy signer status states

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces a new `SessionManagerState` enum and refactors the `SessionManager` class in the Alchemy project.

### Detailed summary
- Added a `SessionManagerState` enum with different states
- Refactored the `SessionManager` class to use the new enum for status management
- Removed event emitting and replaced with state updates in the `SessionManager` class

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/alchemy/src/signer/session/manager.ts [hunk 1]
```diff
@@ -139,9 +149,39 @@ export class SessionManager {
     event: E,
     listener: SessionManagerEvents[E]
   ) => {
-    this.eventEmitter.on(event, listener as any);
-
-    return () => this.eventEmitter.removeListener(event, listener as any);
```
Comment (denniswon) [inline packages/alchemy/src/signer/session/manager.ts:144]: isn't it a breaking change to remove this event emitter though?
The devs could be listening/relying to these events right?

File: packages/alchemy/src/signer/signer.ts [hunk 2]
```diff
@@ -381,14 +379,6 @@ export class AlchemySigner
       });
     });
 
-    this.sessionManager.on("initialized", () => {
-      this.store.setState((state) => ({
-        status: state.user
-          ? AlchemySignerStatus.CONNECTED
-          : AlchemySignerStatus.DISCONNECTED,
-      }));
-    });
```
Comment (denniswon) [inline packages/alchemy/src/signer/signer.ts:390]: oh I guess the session manager is entirely isolated to be within the SDK only.

File: packages/alchemy/src/signer/session/manager.ts (hunk 1 above)
Comment (denniswon) [reply packages/alchemy/src/signer/session/manager.ts:144]: oh I guess the session manager is entirely isolated to be within the SDK only.

**Review by denniswon — APPROVED** [review]
approved but left questions

What was the customer/user issue that was being experienced? @moldy530

**Comment by moldy530** [comment]
ok this wasn't an improvement at all but actually broke shit. so I'm gonna hold off on merging this

**Comment by moldy530** [comment]
I'm gonna close this and take another pass after diagraming some things out

**Pull request closed without merging**